### ISSUE TITLE
BunnyCDN being blocked despite having the compat_bunnycdn

### ISSUE THREAD

**Issue by harryqt** [CIDRAM/CIDRAM#134]
Caught this today:

```
ID: xx-xx-xx
Script Version: CIDRAM v1.13.0
Date/Time: Wed, 10 Jul 2019 00:08:47
IP Address: 66.11.126.218
Hostname: edge-493.b-cdn.net
Referrer: xxxxxxxxxxx
Signatures Count: 1
Signatures Reference: functions.php:L1458
Why Blocked: Fake Googlebot!
User Agent: Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)
Reconstructed URI: xxxxxxxxxxx
reCAPTCHA State: Enabled.
```

The IP belongs to BunnyCDN which I use. I do have **compat_bunnycdn.php** active and the IP is also present their [edgeserverlist](https://bunnycdn.com/api/system/edgeserverlist).

Despite that, the IP were blocked, why? Because the user-agent had  `Googlebot/2.1` present.

First I tested the IP without any user-agent, the result were positive.

![chrome_2019-07-10_01-44-34](https://user-images.githubusercontent.com/18257605/60919853-642d1580-a2b4-11e9-80e7-ee6b88f4ed60.png)

Then with the user-agent: `Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)`

and it was blocked...!!

![chrome_2019-07-10_01-46-37](https://user-images.githubusercontent.com/18257605/60919930-93438700-a2b4-11e9-8391-6ad0cb378570.png)

So, basically cidram not taking the **compat_bunnycdn** module as high priority.

**Comment by ghost** [comment]
That's not bunny-cdn being blocked. That's a block on a fake Googlebot. Totally different case.

**Comment by harryqt** [comment]
Yes, I know.
But the the bunnycdn compatibility module should’ve allow it it regardless the user-agent.

**Comment by ghost** [comment]
Check the loading order defined in configuration->signatures->modules if you want another priority order.

**Comment by Maikuolan** [comment]
Looking at how BunnyCDN actually works, I'm thinking I might need to conditionally disable search engine verification and social media verification entirely, for when the BunnyCDN module is enabled and the request's originating IP address matches one of BunnyCDN's IP addresses (i.e., BunnyCDN might be forwarding on requests that actually originate from Google et al, effectively making reliable search engine verification and social media verification impossible for those forwarded requests). I'll get a patch out shortly.

**Comment by harryqt** [comment]
Yes. If requested IP matches BunnyCDN IPs, then whitelist that request without further checks.

**Comment by Maikuolan** [comment]
Done. :-)

Updating to the latest available versions of CIDRAM and the BunnyCDN module should fix the problem.

**Comment by Maikuolan** [comment]
Scratch that.. found a problem. Sorting it out now. Will reply again in a few minutes.

**Comment by Maikuolan** [comment]
Sorted. :-)

Let me know how it goes.

**Comment by harryqt** [comment]
Everything works perfectly. Thank you so much. Closing it.

**Comment by harryqt** [comment]
Reopening the issue again because of the same type of problem.

I use [httpstatus.io](https://httpstatus.io/), to see headers and some other useful things all the times. I've whitelisted their hosting ip so cidram won't block it. It worked fine until I tried their fetch as Googlebot option.

So, basically they set the User-Agent as Googlebot or other search engines and try to retrieve the page. But now cidram blocking it.

what i don't understand why cidram blocking the request when I already whitelisted the IP?

I thought Whitelist takes highest priority or it's not the case?

![chrome_2019-07-15_08-40-37](https://user-images.githubusercontent.com/18257605/61193979-4f011e00-a6dc-11e9-9970-866b229e15db.png)
![chrome_2019-07-15_08-40-24](https://user-images.githubusercontent.com/18257605/61193982-51fc0e80-a6dc-11e9-8d53-7390d4913f84.png)
